PhysicsTools/Heppy/python/physicsobjects/Tau.py

```diff
@@ -44,6 +48,70 @@ def dxy_approx(self, vertex=None):
         return ( - (vtx.x()-vertex.position().x()) *  p4.y()
                  + (vtx.y()-vertex.position().y()) *  p4.x() ) /  p4.pt()
 
+    def countWP(self,name, allWP=False):
+        '''Returns the number of Working Points 
+        that are passed for given isolation name. 
+        '''
+        WPs = ['VVTight', 'VTight', 'Tight', 'Medium',
+               'Loose', 'VLoose', 'VVLoose', ]
+        testname = 'by{}'+name[2:]
+        n_WP = 0
+        for WP in WPs:
+            
+            # if from heppy-implemented isolation (known list of WPs)
+            if allWP:
+                if not self.tauID(testname.format(WP)):
+                    n_WP += 1 # in this case n_WP is the number of not passed WP
+                else:
+                    return len(WPs) - n_WP #stops counting when reached a passed WP
+
+            # else all WPs will be tested from MINIAOD
+            elif self.physObj.isTauIDAvailable(testname.format(WP)):
+                if self.tauID(testname.format(WP)):
+                    n_WP += 1 # in this case n_WP is the number of passed WP
```

Comment:
do you mean that the returned value is different for the two modes? this cannot be.